packages/fixed-decimal/src/FixedDecimal.test.ts

```diff
@@ -0,0 +1,511 @@
+    })
+    it('works on whole numbers when precision is less than digits', function () {
+      expect(new FixedDecimal('123').toPrecision(2)).toBe('120')
+    })
+    it('works on whole numbers when precision is more than digits', function () {
+      expect(new FixedDecimal('123').toPrecision(6)).toBe('123.000')
+    })
+  })
+
+  describe('toFixed', function () {
+    it('rounds up positive numbers when appropriate', function () {
+      expect(new FixedDecimal('1234.5678').toFixed(2)).toBe('1234.57')
+      expect(new FixedDecimal('1234.5678').toFixed()).toBe('1235')
+    })
+    it('rounds down positive numbers when appropriate', function () {
+      expect(new FixedDecimal('8765.4321').toFixed(2)).toBe('8765.43')
+      expect(new FixedDecimal('8765.4321').toFixed()).toBe('8765')
+    })
+    it('rounds up negative numbers when appropriate', function () {
+      expect(new FixedDecimal('-1234.5678').toFixed(2)).toBe('-1234.57')
+      expect(new FixedDecimal('-1234.5678').toFixed()).toBe('-1235')
+    })
+    it('rounds down negative numbers when appropriate', function () {
+      expect(new FixedDecimal('-8765.4321').toFixed(2)).toBe('-8765.43')
+      expect(new FixedDecimal('-8765.4321').toFixed()).toBe('-8765')
+    })
+    it('has no decimal places when called without args', function () {
+      expect(new FixedDecimal('1', 5).toFixed()).toBe('1')
+    })
+    it('has as many decimal places as specified', function () {
+      expect(new FixedDecimal('1', 5).toFixed(3)).toBe('1.000')
+    })
+    it('adds additional decimal places if beyond the precision of the FixedDecimal', function () {
+      expect(new FixedDecimal('1').toFixed(3)).toBe('1.000')
+    })
+    it('throws when using negative decimal places', function () {
+      expect(() => new FixedDecimal('1').toFixed(-1)).toThrow(
+        'decimalPlaces must be non-negative'
+      )
+    })
+  })
+
+  describe('toShorthand', function () {
+    it('shows zero correctly', function () {
+      expect(new FixedDecimal('0.00000').toShorthand()).toBe('0')
+    })
+    it('shows whole numbers correctly', function () {
+      expect(new FixedDecimal('1234').toShorthand()).toBe('1234')
+    })
+    it('trunctes to whole numbers correctly', function () {
+      expect(new FixedDecimal('8.0099').toShorthand()).toBe('8')
+    })
+    it('shows negative numbers correctly', function () {
+      expect(new FixedDecimal('-8.0099').toShorthand()).toBe('-8')
+    })
+    it('shows decimal numbers correctly', function () {
+      expect(new FixedDecimal('12.345').toShorthand()).toBe('12.34')
+    })
+    it('shows trailing zero if necessary', function () {
+      expect(new FixedDecimal('12.301').toShorthand()).toBe('12.30')
+      expect(new FixedDecimal('12.30').toShorthand()).toBe('12.30')
+    })
+    it('shows thousands correctly', function () {
+      expect(new FixedDecimal('9999.9999').toShorthand()).toBe('9999.99')
+    })
+    it('shows ten thousands correctly', function () {
```

Comment:
nice. should we add some for cases like 
-10000 -> -10k
-9999.99 -> -9999.99

Reply:
nice bug catch!